src/actinia_statistic_plugin/vector_sampling.py

```diff
@@ -0,0 +1,233 @@
+        },
+        "400": {
+            "description": "The error message and a detailed log why vector sampling did not succeed",
+            "schema": ProcessingErrorResponseModel,
+        },
+    },
+}
+
+
+class AsyncEphemeralVectorSamplingResource(ResourceBase):
+    """Perform vector map sampling on a vector map layer based on input points, asynchronous call"""
+
+    decorators = [log_api_call, auth.login_required]
+
+    def _execute(self, location_name, mapset_name, vector_name):
+
+        rdc = self.preprocess(
+            has_json=True,
+            has_xml=False,
+            location_name=location_name,
+            mapset_name=mapset_name,
+            map_name=vector_name,
+        )
+        if rdc:
+            enqueue_job(self.job_timeout, start_job, rdc)
+
+        return rdc
+
+    @swagger.doc(deepcopy(SCHEMA_DOC))
+    def post(self, location_name, mapset_name, vector_name):
+        """Perform vector map sampling on a vector map layer based on input points asynchronously"""
+        self._execute(location_name, mapset_name, vector_name)
+        html_code, response_model = pickle.loads(self.response_data)
+        return make_response(jsonify(response_model), html_code)
+
+
+class SyncEphemeralVectorSamplingResource(AsyncEphemeralVectorSamplingResource):
+    """Perform vector map sampling on a vector map layer based on input points, synchronous call"""
+
+    decorators = [log_api_call, auth.login_required]
+
+    @swagger.doc(deepcopy(SCHEMA_DOC))
+    def post(self, location_name, mapset_name, vector_name):
+        """Perform vector map sampling on a vector map layer based on input points synchronously"""
+        check = self._execute(location_name, mapset_name, vector_name)
+        if check is not None:
+            http_code, response_model = self.wait_until_finish()
+        else:
+            http_code, response_model = pickle.loads(self.response_data)
+        return make_response(jsonify(response_model), http_code)
+
+
+def start_job(*args):
+    processing = AsyncEphemeralVectorSampling(*args)
+    processing.run()
+
+
+class AsyncEphemeralVectorSampling(EphemeralProcessing):
+    """Sample a vector map at vector points"""
+
+    def __init__(self, *args):
+        EphemeralProcessing.__init__(self, *args)
+        self.response_model_class = VectorSamplingResponseModel
+
+    def _execute(self):
+
+        self._setup()
+
+        # Points are stored in self.request_data
+        vector_name = self.map_name
+        points = self.request_data["points"]
+
+        if not points or len(points) == 0:
+            raise AsyncProcessError("Empty coordinate list")
+
+        coordinates_string = ""
+        for tuple in points:
+            if len(tuple) != 3:
+                raise AsyncProcessError("Wrong number of coordinate entries")
+
+            id, x, y = tuple
+            coordinates_string += "%s,%s," % (x, y)
+
+        # g.region probably not needed
```

Comment:
Would be nice to have follow-up ticket to check this later and see if this is not needed.

Reply:
It is used for the vector extent printing.